httpd-client/lib/project/patch.ts

```diff
@@ -113,37 +113,97 @@ export const patchSchema = object({
   title: string(),
   state: patchStateSchema,
   target: string(),
-  tags: array(string()),
+  labels: array(string()),
   merges: array(mergeSchema),
-  reviewers: array(string()),
+  assignees: array(string()),
   revisions: array(revisionSchema),
 }) satisfies ZodSchema<Patch>;
 
 export const patchesSchema = array(patchSchema) satisfies ZodSchema<Patch[]>;
 
+export type LifecycleState =
+  | { status: "draft" }
+  | { status: "open" }
+  | { status: "archived" };
+
+export type CodeLocation = {
+  path: string;
+  old?: { start: number; end: number };
+  new?: { start: number; end: number };
```

Comment:
This has changed actually. I'll update the ts schema.

```rust
/// Code range.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", tag = "type")]
pub enum CodeRange {
    /// One or more lines.
    Lines { range: Range<usize> },
    /// Character range within a line.
    Chars { line: usize, range: Range<usize> },
}

/// Code location, used for attaching comments to diffs.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CodeLocation {
    /// Path of file.
    pub path: PathBuf,
    /// Line range on old file. `None` for added files.
    pub old: Option<CodeRange>,
    /// Line range on new file. `None` for deleted files.
    pub new: Option<CodeRange>,
}
```

Reply:
b2fc41e6a23e5d08037fa931e0248fe2b1a1f691

Reply:
b2fc41e6a23e5d08037fa931e0248fe2b1a1f691